### DISCUSSION TITLE
2021/07/10/compaction

### DISCUSSION THREAD

**Discussion by giscus** [wanghenshui/wanghenshui.github.io#90] [Q&A]
# 2021/07/10/compaction

https://wanghenshui.github.io/2021/07/10/compaction.html

**Comment by zjs1224522500** [comment]
所以 BlobDB 的 GC 是不是相当于完全不会做 rewrite，直接用了一个类似于引用计数的方式来决定是否会删除文件。这样做空间放大应该会很大吧~

**Reply by zjs1224522500** [reply]
看到一个 CSDN 的博主 https://blog.csdn.net/Z_Stand/article/details/117621418 的这篇文章里说 BlobDB 是有 BlobFIle 的文件重写的，但是我看了代码后，跟你的观点是一样的，根本没有做重写，基本都是依赖 SST 失效之后再去回收的相应的 Blob 吧。

**Reply by wanghenshui** [reply]
是
